### PR TITLE
Fix go vet and golint warnings

### PR DESCRIPTION
**Pull request by idealhack**
**What type of PR is this?**

/kind cleanup

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go [hunk 1]
```diff
@@ -282,6 +290,7 @@ const percentErrMsg string = "a valid percent string must be a numeric string fo
 
 var percentRegexp = regexp.MustCompile("^" + percentFmt + "$")
 
+// IsValidPercent checks that a string conforms is a valid percent
```
Comment (ash2k) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: I'm not sure this message and the one below are grammatically correct.

Comment (idealhack) [reply staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: Me too... waiting for a native speaker.

Comment (ash2k) [reply staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: Drop `conforms`?

Comment (cheftako) [reply staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: IsValidPercent checks that a string is in the form of a percentage

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go [hunk 2]
```diff
@@ -391,13 +400,13 @@ func hasChDirPrefix(value string) []string {
 	return errs
 }
 
-// IsSocketAddr checks that a string conforms is a valid socket address
+// IsValidSocketAddr checks that a string conforms is a valid socket address
```
Comment (cheftako) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: IsValidSocketAddr checks that string represents a valid socket address

**Review by cheftako — COMMENTED** [review]
Fix a few minor comments but looks good to me.

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go (hunk 1 above)
Comment (idealhack) [reply staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: @cheftako @ash2k Thank you

**Comment by idealhack** [comment]
@kubernetes/sig-api-machinery-pr-reviews PTAL, thanks

**Comment by cblecker** [comment]
/approve
for `hack/`

**Pull request merged**